hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/FieldProbe.java

```diff
@@ -71,14 +75,14 @@ private String getName(String namePrefix) {
             throw new IllegalArgumentException(format("@Probe field '%s' is of an unhandled type", field));
         }
 
-        if (AccessibleObjectProbe.isDouble(type)) {
+        if (isDouble(type)) {
             return new DoubleFieldProbe<S>(field, probe, type);
```

Comment:
it seems it's both a factory and superclass of implementations produced by the factory. 
as I said - I'm not sure if that's really a problem, but it smells, to me anyway.